File: lua/two-fer/two-fer.lua
--@module TwoFer 
local TwoFer = {}

--- Output message format
local message_format <const> = "One for %s, one for me."
--- 
local default_pronoun <const> = "you"

--- Returns a message in the format "One for <name>, one for me". If no name
--- is provided, the person is referred to as "you".
-- @param name string? Person's name
-- @return string Formatted message 
function TwoFer.two_fer(name)
    local pronoun = name or default_pronoun

    return string.format(message_format, pronoun)
end

return TwoFer

File: csharp/two-fer/TwoFer.cs
public static class TwoFer
{
    /// <summary>
    /// Returns a message to greet a person when you give them the cookie.
    /// If you know their name, refer to them by their name, otherwise
    /// refer to them as "you".
    /// </summary>
    /// <param name="personName">Name of person, if known.</param>
    /// <returns>Message in the format "One for [person], one for me."</returns>
    public static string Speak(string personName = "you")
        => $"One for {personName}, one for me.";
}
